hooks/read-package-hook/src/createPeerDependencyPatcher.ts

```diff
@@ -44,6 +46,22 @@ export function createPeerDependencyPatcher (
 
       pkg.peerDependencies![peerName] = currentVersions.join(' || ')
     }
+
+    let overrides: ReturnType<typeof parseOverrides>
+    try {
+      overrides = parseOverrides(peerDependencyRules.allowedVersions ?? {})
+    } catch (e) {
+      const selector = (e as PnpmError)?.message?.match(/".*"/)?.[0] ?? ''
+      throw new PnpmError('INVALID_ALLOWED_VERSION_SELECTOR', `Cannot parse the ${selector} selector in the allowedVersions`)
```

Comment:
let's create a new error class for this error and put the selector to a separate field. This way you want need to parse the error message to get the selector.